workloads/simpleinsert1/simpleinsert1.go

```diff
@@ -58,16 +60,27 @@ var wl = &workload.Workload{
 
 		cols := parseColumns(columns)
 		columnList := strings.Join(cols, ",")
-		data := randStringArray(len(cols), dataLen)
-		args := make([]any, len(data))
-		for i, v := range data {
-			args[i] = v
-		}
 
-		q := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", tableName, columnList, strings.Repeat("?,", len(cols)-1)+"?")
-		_, err = tx.ExecContext(ctx, q, args...)
-		if err != nil {
-			return fmt.Errorf("in exec for query: %q, error: %w", q, err), false
+		for i := 0; i < insertsPerTx; i++ {
+			data := randStringArray(len(cols), dataLen)
+			args := make([]any, len(data))
+			for i, v := range data {
+				args[i] = v
+			}
+
+			q := fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", tableName, columnList, strings.Repeat("?,", len(cols)-1)+"?")
+			_, err = tx.ExecContext(ctx, q, args...)
+			if err != nil {
+				if insertsPerTx > 1 && strings.Contains(err.Error(), "transaction rolled back to reverse changes") {
```

Comment:
Can you just return the error instead of opening a new transaction here? It will be rolled back and retried.

Reply:
done